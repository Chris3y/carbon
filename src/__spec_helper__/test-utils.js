const isUpper = char => char.toUpperCase() === char;
const humpToDash = (acc, char) => `${acc}${isUpper(char) ? `-${char.toLowerCase()}` : char}`;

export const toCSSCase = (str) => {
  return str.split('').reduce(humpToDash, '');
};

export const assertStyleMatch = (styleSpec, component) => {
  Object.entries(styleSpec).forEach(([attr, value]) => {
    expect(component).toHaveStyleRule(toCSSCase(attr), value);
  });
};
