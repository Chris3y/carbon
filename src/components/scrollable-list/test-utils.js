import React from 'react';
import ScrollableListItem from './scrollable-list-item.component';

const makeArray = n => [...Array(n).keys()];

const listItemReducer = ({ nonSelectables = [], customSelectables = [] }) => {
  // generate jsx for selectable list items, based on indexes in config
  return (acc, item, index) => {
    if (nonSelectables.includes(item)) return [...acc, <div />];
    if (customSelectables.includes(item)) return [...acc, <div isSelectable />];

    return [...acc, <ScrollableListItem id={ index } />];
  };
};

const renderListItems = (opts) => {
  return makeArray(opts.num).reduce(listItemReducer(opts), []);
};


const keyPress = (code) => {
  const ev = new KeyboardEvent('keydown', { which: code });
  document.dispatchEvent(ev);
};

const keyMap = {
  UpArrow: '38',
  DownArrow: '40',
  RightArrow: '39',
  LeftArrow: '37',
  Enter: '13',
  Tab: '9'
};

const repeat = action => (n = 1) => makeArray(n).forEach(() => action());

const keyboard = Object.keys(keyMap).reduce((acc, key) => {
  acc[`press${key}`] = () => repeat(keyPress(keyMap[key]));
  return acc;
}, {});

const listFrom = wrapper => wrapper.find('ul');
const childrenFrom = node => node.children();

const hoverList = wrapper => (item) => {
  childrenFrom(listFrom(wrapper)).at(item).simulate('mouseover');
};

const simulateEvent = eventName => wrapper => wrapper.simulate(eventName);
const click = simulateEvent('click');

const selectedItemOf = wrapper => wrapper.state().selectedItem;
const isUnique = (val, index, self) => self.indexOf(val) === index;
const isSelectableGiven = nonSelectables => i => !nonSelectables.includes(i);

const selectedItemReducer = method => wrapper => (acc, i) => {
  method(wrapper)(i);
  return [...acc, selectedItemOf(wrapper)];
};

const arraysEqual = (arr1, arr2) => arr1.sort().join(',') === arr2.sort().join(',');

const assertCorrectTraversal = method => expect => ({ num, nonSelectables = [] }) => (wrapper) => {
  const array = makeArray(num);
  const validIndexes = array.filter(isSelectableGiven(nonSelectables));

  const indexesThatWereSelected = array
    .reduce(selectedItemReducer(method)(wrapper), [])
    .filter(isUnique);
  expect(arraysEqual(validIndexes, indexesThatWereSelected)).toBeTruthy();
};

const assertKeyboardTraversal = assertCorrectTraversal(() => keyboard.pressDownArrow)(expect);
const assertHoverTraversal = assertCorrectTraversal(wrapper => hoverList(wrapper))(expect);

export {
  hoverList,
  selectedItemOf,
  childrenFrom,
  renderListItems,
  keyboard,
  assertKeyboardTraversal,
  assertHoverTraversal,
  listFrom,
  click
};
