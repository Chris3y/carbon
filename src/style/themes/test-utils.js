import generatePalette from '../palette';
import atOpacity from '../utils/at-opacity';
import baseColors from '../color-config';

const addHex = base => `#${base}`;

export const palette = {
  ...generatePalette(baseColors),
  atOpacity,
  ...Object.keys(baseColors).reduce((acc, col) => {
    acc[col] = addHex(baseColors[col]);
    return acc;
  }, {})
};
const blackWithOpacity = atOpacity('#000000');

export const baseThemeConfig = {
  colors: {
    brand: palette.brilliantGreen,
    white: '#FFFFFF',
    error: palette.errorRed,
    warning: palette.gold,
    success: palette.brilliantGreenShade(20),
    info: palette.productBlueShade(3),
    text: {
      body: blackWithOpacity(0.9),
      disabled: blackWithOpacity(0.55),
      placeholder: blackWithOpacity(0.3)
    }
  }
};


const isObject = (obj) => {
  return typeof obj === 'object' && obj !== null;
};

export const assertIsSubset = (obj, comparison) => {
  if (!isObject(obj)) {
    // no further nesting, assert that values are equal
    expect(obj).toEqual(comparison);
    return;
  }

  const objKeys = Object.keys(obj);
  const comparisonKeys = Object.keys(comparison);

  objKeys.forEach((key) => {
    // assert that keys are present
    expect(comparisonKeys.includes(key)).toBeTruthy();

    // repeat for nested objects
    assertIsSubset(obj[key], comparison[key]);
  });
};
