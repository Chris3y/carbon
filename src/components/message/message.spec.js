import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import OptionsHelper from '../../utils/helpers/options-helper/options-helper';
import MessageStyle from './message.style';
import Message from './message.component';
import classicTheme from '../../style/themes/classic';
import CloseIcon from './close-icon/close-icon.component';
import { assertStyleMatch } from '../../__spec_helper__/test-utils';

function render(props) {
  return TestRenderer.create(<MessageStyle { ...props }>Message</MessageStyle>);
}

describe('Message', () => {
  describe('when in small/medium/large theme', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Message>Message</Message>);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('does not render the close icon when onDismiss prop is not provided', () => {
      const closeIcon = wrapper.find(CloseIcon);
      expect(closeIcon.exists()).toEqual(false);
    });

    it('renders the close icon when onDismiss function is provided', () => {
      const onDismiss = jest.fn();
      wrapper.setProps({ onDismiss });
      const closeIcon = wrapper.find(CloseIcon);
      expect(closeIcon.exists()).toEqual(true);
    });
  });

  describe('when transparent prop is set to true', () => {
    it('should render the message without the border', () => {
      const wrapper = render({
        transparent: true,
        type: 'info'
      });

      assertStyleMatch(
        {
          border: 'none'
        },
        wrapper.toJSON()
      );
    });
  });

  describe('when transparent prop is not passed', () => {
    it('should render the message with border in a proper color and a white background', () => {
      OptionsHelper.messages.forEach((variant) => {
        const wrapper = render({ variant });
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('when in classic mode', () => {
    describe('when rendered', () => {
      it('should match the snapshot', () => {
        OptionsHelper.colors.forEach((variant) => {
          const wrapper = render({ theme: classicTheme, variant });
          expect(wrapper).toMatchSnapshot();
        });
      });
    });

    describe('when transparent prop is set to true', () => {
      it('should render the message without the border and with background transparent', () => {
        const wrapper = render({
          transparent: true,
          theme: classicTheme,
          variant: 'info'
        });

        assertStyleMatch(
          {
            border: 'none',
            backgroundColor: 'transparent'
          },
          wrapper.toJSON()
        );
      });
    });

    describe('when border prop is set to false', () => {
      it('should render the message without a border', () => {
        const wrapper = render({
          border: false,
          theme: classicTheme,
          variant: 'info'
        });

        assertStyleMatch(
          {
            border: 'none'
          },
          wrapper.toJSON()
        );
      });
    });

    describe('when roundedCorners prop is set to false', () => {
      it('should apply no border-radius style', () => {
        const wrapper = render({
          roundedCorners: false,
          theme: classicTheme,
          variant: 'info'
        });

        assertStyleMatch(
          {
            borderRadius: '0px'
          },
          wrapper.toJSON()
        );
      });
    });

    describe('when roundedCorners prop is not passed', () => {
      it('should apply proper border-radius style', () => {
        const wrapper = render({
          theme: classicTheme,
          type: 'info'
        });

        assertStyleMatch(
          {
            borderRadius: '3px'
          },
          wrapper.toJSON()
        );
      });
    });
  });
});
