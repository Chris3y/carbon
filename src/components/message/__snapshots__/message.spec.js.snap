// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message when in classic mode when border prop is set to false should render the message without a border 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid;
  border: none;
  background-color: #335c6d;
}

<info
  className="c0"
  type="default"
>
  Message
</info>
`;

exports[`Message when in classic mode when border prop is set to false should render the message without a border 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #C7384F;
  border: none;
  background-color: #fdf5f5;
}

<info
  className="c0"
  type="error"
>
  Message
</info>
`;

exports[`Message when in classic mode when border prop is set to false should render the message without a border 3`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid;
  border: none;
  background-color: #fffbf2;
}

<info
  className="c0"
  type="help"
>
  Message
</info>
`;

exports[`Message when in classic mode when border prop is set to false should render the message without a border 4`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #0073C2;
  border: none;
  background-color: #f3f8fe;
}

<info
  className="c0"
  type="info"
>
  Message
</info>
`;

exports[`Message when in classic mode when border prop is set to false should render the message without a border 5`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid;
  border: none;
  background-color: #fff8f2;
}

<info
  className="c0"
  type="maintenance"
>
  Message
</info>
`;

exports[`Message when in classic mode when border prop is set to false should render the message without a border 6`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid;
  border: none;
  background-color: #f7f5fa;
}

<info
  className="c0"
  type="new"
>
  Message
</info>
`;

exports[`Message when in classic mode when border prop is set to false should render the message without a border 7`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #00B000;
  border: none;
  background-color: #f6fbf6;
}

<info
  className="c0"
  type="success"
>
  Message
</info>
`;

exports[`Message when in classic mode when border prop is set to false should render the message without a border 8`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #E96400;
  border: none;
  background-color: #fff8f2;
}

<info
  className="c0"
  type="warning"
>
  Message
</info>
`;

exports[`Message when in classic mode when rendered should match the snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #C7384F;
  border: 1px solid #f7d9d9;
  background-color: #fdf5f5;
}

<info
  className="c0"
  type="error"
>
  Message
</info>
`;

exports[`Message when in classic mode when rendered should match the snapshot 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #0073C2;
  border: 1px solid #d0e3fa;
  background-color: #f3f8fe;
}

<info
  className="c0"
  type="info"
>
  Message
</info>
`;

exports[`Message when in classic mode when rendered should match the snapshot 3`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #00B000;
  border: 1px solid #dcf1da;
  background-color: #f6fbf6;
}

<info
  className="c0"
  type="success"
>
  Message
</info>
`;

exports[`Message when in classic mode when rendered should match the snapshot 4`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #E96400;
  border: 1px solid #ffe4cc;
  background-color: #fff8f2;
}

<info
  className="c0"
  type="warning"
>
  Message
</info>
`;

exports[`Message when in classic mode when transparent prop is set to true should render the message without the border and background color 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
  border: none;
}

<info
  className="c0"
  type="default"
>
  Message
</info>
`;

exports[`Message when in classic mode when transparent prop is set to true should render the message without the border and background color 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
  border: none;
}

<info
  className="c0"
  type="error"
>
  Message
</info>
`;

exports[`Message when in classic mode when transparent prop is set to true should render the message without the border and background color 3`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
  border: none;
}

<info
  className="c0"
  type="help"
>
  Message
</info>
`;

exports[`Message when in classic mode when transparent prop is set to true should render the message without the border and background color 4`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
  border: none;
}

<info
  className="c0"
  type="info"
>
  Message
</info>
`;

exports[`Message when in classic mode when transparent prop is set to true should render the message without the border and background color 5`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
  border: none;
}

<info
  className="c0"
  type="maintenance"
>
  Message
</info>
`;

exports[`Message when in classic mode when transparent prop is set to true should render the message without the border and background color 6`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
  border: none;
}

<info
  className="c0"
  type="new"
>
  Message
</info>
`;

exports[`Message when in classic mode when transparent prop is set to true should render the message without the border and background color 7`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
  border: none;
}

<info
  className="c0"
  type="success"
>
  Message
</info>
`;

exports[`Message when in classic mode when transparent prop is set to true should render the message without the border and background color 8`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
  border: none;
}

<info
  className="c0"
  type="warning"
>
  Message
</info>
`;

exports[`Message when rendered should match the snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #C7384F;
}

<info
  className="c0"
  type="error"
>
  Message
</info>
`;

exports[`Message when rendered should match the snapshot 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #0073C2;
}

<info
  className="c0"
  type="info"
>
  Message
</info>
`;

exports[`Message when rendered should match the snapshot 3`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #00B000;
}

<info
  className="c0"
  type="success"
>
  Message
</info>
`;

exports[`Message when rendered should match the snapshot 4`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid #E96400;
}

<info
  className="c0"
  type="warning"
>
  Message
</info>
`;

exports[`Message when roundedCorners prop is not passed should apply proper border-radius style 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  background-color: #FFFFFF;
  border: 1px solid;
}

<info
  className="c0"
>
  Message
</info>
`;

exports[`Message when roundedCorners prop is set to false should apply no border-radius style 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: none;
  background-color: #FFFFFF;
  border: 1px solid;
}

<info
  className="c0"
>
  Message
</info>
`;

exports[`Message when transparent prop is set to true should render the message without the border 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
}

<info
  className="c0"
  type="default"
>
  Message
</info>
`;

exports[`Message when transparent prop is set to true should render the message without the border 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
}

<info
  className="c0"
  type="error"
>
  Message
</info>
`;

exports[`Message when transparent prop is set to true should render the message without the border 3`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
}

<info
  className="c0"
  type="help"
>
  Message
</info>
`;

exports[`Message when transparent prop is set to true should render the message without the border 4`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
}

<info
  className="c0"
  type="info"
>
  Message
</info>
`;

exports[`Message when transparent prop is set to true should render the message without the border 5`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
}

<info
  className="c0"
  type="maintenance"
>
  Message
</info>
`;

exports[`Message when transparent prop is set to true should render the message without the border 6`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
}

<info
  className="c0"
  type="new"
>
  Message
</info>
`;

exports[`Message when transparent prop is set to true should render the message without the border 7`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
}

<info
  className="c0"
  type="success"
>
  Message
</info>
`;

exports[`Message when transparent prop is set to true should render the message without the border 8`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-radius: 3px;
  border: none;
  background-color: #FFFFFF;
}

<info
  className="c0"
  type="warning"
>
  Message
</info>
`;
